simulations.py

```diff
@@ -64,42 +75,73 @@ def w_fun(CiT,Ui):
 def inv_nla_jit(A):
   return np.linalg.inv(A)
 
-def update_Ui(Cit,Ui,mu_Ui, Sigma_Ui, Nset):
+def update_Ui(Cit, Ui, ce_Ui, Sigma_Ui, Nset):
+    """ Update beliefs using Baysian updating on Normal Distribution
+    Args:
+        Cit () :
+        Ui () :
+        ce_Ui (numpy.ndarray) : certainty equivelents. previously mu_Ui. Shape (N,)
+        Sigma_Ui () :
+        Nset (range) :
+    """ 
+    
+    # https://en.wikipedia.org/wiki/Multivariate_normal_distribution#Conditional_distributions
+    # μ_bar = μ_1 + Ε12 Ε22^-1 ( a - μ_2 )  
+
     x1 = Cit
     x2 = [n for n in Nset if n not in Cit]
     Nit = [n for n in Nset if n not in Cit]
-    mu1 = np.array([mu_Ui[n] for n in x1]).reshape((1,len(x1)))
-    mu2 = np.array([mu_Ui[n] for n in x2]).reshape((1,len(x2)))
+
+    mu1 = np.array([ce_Ui[n] for n in x1]).reshape((1,len(x1)))
+    mu2 = np.array([ce_Ui[n] for n in x2]).reshape((1,len(x2)))
+
     Sigma11 = np.ones((len(x1),len(x1)))
     Sigma21 = np.ones((len(x2),len(x1)))
+
     for i in range(len(Cit)):
         for j in range(len(Cit)):
-            Sigma11[i,j] = Sigma_Ui[Cit[i],Cit[j]]
+            Sigma11[i,j] = Sigma_Ui[Cit[i],Cit[j]] 
+
     for i in range(len(Cit)):
         for j in range(len(Nit)):
-            Sigma21[j,i] = Sigma_Ui[Cit[i], Nit[j]]
+            Sigma21[j,i] = Sigma_Ui[Cit[i], Nit[j]] 
+
     a = np.array([Ui[n] for n in x1]).reshape((1,len(x1)))
     inv_mat = inv_nla_jit(Sigma11)
     inner = np.matmul(Sigma21, inv_mat)
     mubar = mu2 + (np.matmul(inner,(a-mu1).T)).T
-    mu_new = Ui
+    ce_new = Ui
+
     for i in range(len(x2)):
-        mu_new[x2[i]] = mubar[0,i]
-    return mu_new
+        ce_new[x2[i]] = mubar[0,i]
+    return ce_new
 
 ## CHOICE FUNCTIONS
-def choice_helper(Cit,mu, Nset):
-    x2 = [n for n in Nset if n not in Cit]
-    cit = x2[np.argmax([mu[i] for i in x2])]
+def choice_helper(Cit,mu, choice_set):
+    """
+    Args:
+        Cit (list):
+        mu (numpy.ndarray):
+        choice_set (list) :
+    """
+    cit = choice_set[np.argmax([mu[i] for i in choice_set])]
     return cit
 
-def choice_ind(U_i,mu_U_i, Sigma_U_i,T,N, Nset):
+def choice_ind(U_i, mu_U_i, Sigma_U_i, T, N, Nset, alpha, epsilon):
     C_iT = []
     for t in range(T):
         mu_Uit = mu_U_i
         if len(C_iT) > 0:
+            # update beliefs
             mu_Uit = update_Ui(C_iT,U_i,mu_U_i, Sigma_U_i, Nset)
```

Comment:
we need to add a definition of Sigma_U_it and return the updated sigmas from the update function